### DISCUSSION TITLE
material ui text field with react hook form to mask input

### DISCUSSION THREAD

**Discussion by sivaputrevu** [react-hook-form/react-hook-form#2393] [Q&A]
Hi All,

I was searching for masking input to block special characters for a text field such as names which shouldn't have special charecters.
I created one example where the name is mandatory field[Material UI text field] and blocks special charecters and allows only alphabets[case doesn't matter].
thought of sharing it!
here is the code sandbox: 
URL: https://codesandbox.io/s/hidden-cache-xjroe?fontsize=14&hidenavigation=1&theme=dark

**Comment by bluebill1049** [comment]
is this what you after: https://github.com/react-hook-form/react-hook-form/discussions/2311#discussioncomment-43038?

**Reply by sivaputrevu** [reply]
I haven't seen that one! I jus added one more field with some custom validation using mask
